lua/telescope/make_entry.lua

```diff
@@ -1157,9 +1157,17 @@ function make_entry.gen_from_diagnostics(opts)
     { remaining = true },
   }
   local line_width = vim.F.if_nil(opts.line_width, 0.5)
+  local line_width_opts = { width = line_width }
+  if type(line_width) == "string" then
+    if line_width == "full" then
+      line_width_opts = {}
+    else
+      error(string.format("'%s' is not a valid value for line_width", line_width))
```

Comment:
I think we should move the validation of `line_width` to declaration of the picker and use `utils.notify` instead of `error`.

Reply:
Hi, thanks for checking my PR :)

I changed error to utils.notify, but left it in the same place. I couldn't find "good" place for validation in diagnostics picker, there is no direct call to line_width opt in here. Also, function 'make_entry.gen_from_diagnostics' is used specifically for diagnostics, so I guess it's not the worst place for this validation.

Reply:
The problem is that unlike with `error`, `utils.notify` will let the picker continue to run and it'll eventually throw an actual error that isn't as helpful. 

I think we can simplify this bit of code to
```lua
  if type(line_width) == "string" and line_width == "full" then
    line_width_opts = {}
  end
```

then in `lua/telescope/builtin/__diagnostics.lua:124`, do like
```lua
  if type(opts.line_width) == "string" and opts.line_width ~= "full" then
    utils.notify("builtin.diagnostics", {
      msg = string.format("'%s' is not a valid value for line_width", opts.line_width),
      level = "ERROR",
    })
    return
  end
```